### PR TITLE
fedekunze/1614-optimistic updates on PageProposal

### LINKED ISSUE #1614: Proposal state not updated after deposit
Description:

Probably related to #1591 or that we are not fetching the updated proposal after a deposit

### PR DESCRIPTION
**Pull request by fedekunze**
Closes #1614

_Description:_

- updates the proposal displayed on `PageProposal` after successful vote and deposit
- added `getProposal(proposalId)` function on store to query a single proposal
- minor bug fixes
- changed `_block` --> `_time` on mocked LCD proposals

❤️ Thank you!

---

* [x] Added entries in `CHANGELOG.md` with issue # and GitHub username
* [x] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: app/src/renderer/components/governance/PageProposal.vue [hunk 1]
```diff
@@ -101,7 +97,25 @@ export default {
...
+    proposal() {
+      let proposal = this.proposals[this.proposalId]
+      if (proposal) {
+        proposal.tally_result.yes = String(
+          Math.round(parseFloat(proposal.tally_result.yes))
+        )
+        proposal.tally_result.no = String(
+          Math.round(parseFloat(proposal.tally_result.no))
+        )
+        proposal.tally_result.no_with_veto = String(
```
Comment (jbibla) [inline app/src/renderer/components/governance/PageProposal.vue]: this is weird, no? does it have to be a string?

`String(Math.round(parseFloat(somethingHere)))`

File: app/src/renderer/components/governance/PageProposal.vue [hunk 2]
```diff
@@ -137,6 +151,41 @@ export default {
...
+          color: `blue`
+        }
+      else
+        return {
+          message: `There was an error determining the status of this proposal.`,
+          color: `grey`
+        }
+    }
+  },
+  watch: {
```
Comment (jbibla) [inline app/src/renderer/components/governance/PageProposal.vue]: why is this necessary?

Comment (jbibla) [inline app/src/renderer/components/governance/PageProposal.vue]: what does this accomplish exactly?

Comment (fedekunze) [reply app/src/renderer/components/governance/PageProposal.vue]: ohh that watch, I can delete it (one of my attempts to make it work)

File: app/src/renderer/components/governance/PageProposal.vue (hunk 1 above)
Comment (fedekunze) [reply app/src/renderer/components/governance/PageProposal.vue]: well not necessarily

Comment (fedekunze) [reply app/src/renderer/components/governance/PageProposal.vue]: fixed

File: app/src/renderer/components/governance/PageProposal.vue (hunk 2 above)
Comment (fedekunze) [reply app/src/renderer/components/governance/PageProposal.vue]: deleted
